packages/opentelemetry-resources/src/platform/node/detect-resources.ts

```diff
@@ -78,9 +74,7 @@ const logResources = (logger: Logger, resources: Array<Resource>) => {
         sorted: true,
         compact: false,
       });
-      const detectorName = DETECTORS[index].constructor
-        ? DETECTORS[index].constructor.name
-        : 'Unknown detector';
+      const detectorName = resource.detectedBy;
```

Comment:
I've modified the `Resource` ctor to keep this functionality, but I don't think my change is ideal... /cc @mwear

Reply:
@mwear would that mean deleting this `detectorName` stuff altogether and letting the detectors handle the detector name stuff?

Reply:
Yeah, but I think you could go one step further, and have the Detector log the detected resources as they're found, rather than finding them all first and logging afterwards. Specifically, you could move this logging into the detector:  https://github.com/open-telemetry/opentelemetry-js/pull/1420/files#diff-1328771aa7fff461aee162025e488e55R67-R79.

Do this only if you think it's an improvement though.